test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -1181,6 +1201,33 @@ static void ipsec_test_capability(void)
 	CU_ASSERT(odp_ipsec_capability(&capa) == 0);
 }
 
+static void test_ipsec_stats(void)
+{
+	ipsec_test_flags flags = {0};
+	uint32_t a, c;
+	int rc = -1;
+
+	flags.stats_success = 1;
+
+	for (c = 0; c < ARRAY_SIZE(ciphers); c++)
+		for (a = 0; a < ARRAY_SIZE(auths); a++) {
+			rc = test_esp_out_in(&ciphers[c], &auths[a], &flags);
+			if (rc == 0)
+				return;
+		}
+
+	for (c = 0; c < ARRAY_SIZE(cipher_auth_comb); c++) {
+		rc = test_esp_out_in(&cipher_auth_comb[c].cipher,
+				     &cipher_auth_comb[c].auth,
+				     &flags);
+		if (rc == 0)
+			return;
+	}
+
+	CU_ASSERT_NOT_EQUAL_FATAL(rc, 0);
```

Comment:
rc cannot ever be zero here.

Reply:
Idea is to make sure we find atleast one supported algo. Let me recheck if we need this.

Reply:
But rc is always -1 if the code ends up here, no matter what is supported and what the ODP implementation does.

Reply:
Yeah. The returns had to be changed. The returns were a temporary thing which I missed. Will have it addressed with next version. Thanks.